app/views/reservations/_reservation_fields.html.haml

```diff
@@ -1,10 +1,12 @@
 .datetime-block
   - start_disabled = start_time_disabled?(f.object)
-  = f.input :reserve_start_date, :input_html => { :class => 'datepicker', disabled: start_disabled }
+  - date_field = f.object.actual_start_at ? :actual_start_date : :reserve_start_date
+  = f.input date_field, :input_html => { :class => 'datepicker', disabled: start_disabled }
```

Comment:
I reworked this slightly because of some bugs when calculating `reserve_end_at`. Also, hacking the params does nothing, so we should be good with that.